src/shipit_code_coverage/tests/test_utils.py

```diff
@@ -57,12 +58,32 @@ def test_threadpoolexecutorresult():
     except:
         assert True
 
+    # Tast that ThreadPoolExecutorResult's context returns as soon as a task fails.
     try:
         now = time.time()
         with utils.ThreadPoolExecutorResult() as executor:
-            executor.submit(lambda: time.sleep(10))
+            executor.submit(lambda: time.sleep(5))
             executor.submit(do_raise)
         assert False
     except:
         assert True
-        assert time.time() - now < 5
+        assert time.time() - now < 2
+
+    # Test that futures that were not scheduled yet are cancelled.
+    try:
+        with utils.ThreadPoolExecutorResult(max_workers=1) as executor:
+            f1 = executor.submit(lambda: time.sleep(1))
+            f2 = executor.submit(do_raise)
+            executor.submit(lambda: time.sleep(1))
+            f4 = executor.submit(lambda: time.sleep(1))
+            f5 = executor.submit(lambda: time.sleep(1))
+            f6 = executor.submit(lambda: time.sleep(1))
+        assert False
+    except:
+        assert True
```

Comment:
Nit: Why the `assert False` and `assert True`?

Reply:
The `assert False` is just to make sure we are throwing the exception (the normal ThreadPoolExecutor doesn't throw an exception when one of the future fails, the custom ThreadPoolExecutorResult is instead supposed to).
The `assert True` is actually uneeded, I will remove it.

Reply:
Removed!